utils/_decorators.py

```diff
@@ -106,109 +94,52 @@ def _should_skip(condition=None, library=None, weblog_variant=None):
     return True
 
 
-def missing_feature(condition: bool = None, library=None, weblog_variant=None, reason=None, force_skip: bool = False):
-    """decorator, allow to mark a test function/class as missing"""
-
-    skip = _should_skip(library=library, weblog_variant=weblog_variant, condition=condition)
-
-    def decorator(function_or_class):
-
-        if inspect.isclass(function_or_class):
-            assert condition is not None or (library is None and weblog_variant is None), _MANIFEST_ERROR_MESSAGE
-
-        if not skip:
-            return function_or_class
-
-        full_reason = "missing_feature" if reason is None else f"missing_feature ({reason})"
-
-        return _get_expected_failure_item(function_or_class, full_reason, force_skip=force_skip)
+def decorator(expected_to_fail, force_skip, condition, decorator_type, reason, function_or_class):
+    if inspect.isclass(function_or_class):
+        assert condition is not None, _MANIFEST_ERROR_MESSAGE
 
-    return decorator
-
-
-def incomplete_test_app(
-    condition: bool = None, library=None, weblog_variant=None, reason=None, force_skip: bool = False
-):
-    """decorator, marks tests that rely on endpoints in test apps that are incomplete or missing"""
-
-    skip = _should_skip(library=library, weblog_variant=weblog_variant, condition=condition)
-
-    def decorator(function_or_class):
+    if decorator_type in ("bug", "flaky"):
+        _ensure_jira_ticket_as_reason(function_or_class, reason)
 
-        if inspect.isclass(function_or_class):
-            assert condition is not None or (library is None and weblog_variant is None), _MANIFEST_ERROR_MESSAGE
+    full_reason = decorator_type if reason is None else f"{decorator_type} ({reason})"
+    if not expected_to_fail:
+        return function_or_class
+    return _get_expected_failure_item(function_or_class, full_reason, force_skip)
 
-        if not skip:
-            return function_or_class
 
-        full_reason = "incomplete_test_app" if reason is None else f"incomplete_test_app ({reason})"
+def missing_feature(condition=None, library=None, weblog_variant=None, reason=None, force_skip: bool = False):
+    """decorator, allow to mark a test function/class as missing"""
+    expected_to_fail = _expected_to_fail(library=library, weblog_variant=weblog_variant, condition=condition)
+    return partial(decorator, expected_to_fail, force_skip, condition, "missing_feature", reason)
 
-        return _get_expected_failure_item(function_or_class, full_reason, force_skip=force_skip)
 
-    return decorator
+def incomplete_test_app(condition=None, library=None, weblog_variant=None, reason=None):
+    """Decorator, allow to mark a test function/class as not compatible with the tested application"""
+    expected_to_fail = _expected_to_fail(library=library, weblog_variant=weblog_variant, condition=condition)
+    force_skip = False
+    return partial(decorator, expected_to_fail, force_skip, condition, "incomplete_test_app", reason)
```

Comment:
Same comment as above : 

```suggestion
    return partial(decorator, expected_to_fail=expected_to_fail, force_skip=False, condition=condition, decorator_type="incomplete_test_app", reason=reason)
```

Reply:
We can only use positional arguments in partial(....)